neofetch

```diff
@@ -3202,6 +3202,49 @@ END
             # Default fallback font hardcoded in terminal-preferences.c
             [[ -z "$term_font" ]] && term_font="Monospace 12"
         ;;
+
+        "conemu-"*)
+            local ce_arg_list
+            local ce_arg_idx
+            local ce_conf
+
+            # Could have used `eval set -- "$ConEmuArgs"` instead for arg parsing but eval is evil...
+            readarray -t ce_arg_list < <(xargs -n1 printf "%s\n" <<< "${ConEmuArgs-}")
+
+            for ce_arg_idx in "${!ce_arg_list[@]}"; do
+                # Search for "-LoadCfgFile" arg
+                [[ "${ce_arg_list[$ce_arg_idx]}" != -LoadCfgFile ]] && continue
+
+                # Conf path is the next arg
+                ((++ce_arg_idx))
+                ce_conf="${ce_arg_list[$ce_arg_idx]}"
+                break
+            done
+
+            # https://conemu.github.io/en/ConEmuXml.html#search-sequence
+            local ce_seq=(
+                "$ce_conf"
+                "$ConEmuDir\ConEmu.xml"
+                "$ConEmuDir\.ConEmu.xml"
+                "$ConEmuBaseDir\ConEmu.xml"
+                "$ConEmuBaseDir\.ConEmu.xml"
+                "$APPDATA\ConEmu.xml"
+                "$APPDATA\.ConEmu.xml"
+            )
+
+            for ce_conf in "${ce_seq[@]}"; do
+                # Search for first conf file available
+                [[ ! -f "$ce_conf" ]] && continue
+
+                # Very basic XML parsing
+                term_font="$(awk '/name="FontName"/ && match($0, /data="([^"]*)"/) {print substr($0, RSTART+6, RLENGTH-7)}' "$ce_conf")"
```

Comment:
There's probably a simpler way of parsing this too. Got an example config I can look at?

Reply:
[Yup](https://github.com/Maximus5/ConEmu/blob/master/Release/ConEmu/ConEmu_Sample.xml).

Reply:
This is one way of doing it in `bash` (same limitations as your method).

```sh
while IFS="= />\'\"" read -ra line; do
    [[ ${line[3]} == FontName ]] &&
        echo "${line[9]}"
done < ~/sample.xml
```

Reply:
Done! This can even parse cases like:

```
            <value
                name='FontName'
                type="string"
                data='Consolas'/>

```

```sh
while IFS=\> read -rd \< xml_data _; do
    [[ $xml_data == *[\"\']FontName[\"\']* ]] && {
        term_font=${xml_data##*=[\"\']}
        term_font=${term_font%%[\"\']/}
        break
    }
done < "$ce_conf"
```

It's fast too.